src/typing.py

```diff
@@ -2002,12 +2009,9 @@ def __new__(cls, typename, bases, ns):
         nm_tpl._field_defaults = defaults_dict
         # update from user namespace without overriding special namedtuple attributes
         for key in ns:
-            if not hasattr(nm_tpl, key):
+            if key not in _prohibited + _special + nm_tpl._fields:
```

Comment:
The tuple concatenation here sounds like it could be slow -- since the 'in' operator still does a linear search, you might as well do separate checks, e.g. `key not in _special and key not in nm_tmp._fields`.